### PR TITLE
Add ESLint

### PR DESCRIPTION
**Pull request by IanVS**
This adds a very basic set of eslint rules, only their core recommended rules, plus a rule to enforce that `node:` prefixes are used.  More rules could be added later, if desired.

I'm opening the PR with some failing linting rules, so we can see that CI works correctly, then I'll fix the errors.

### CODE REVIEW

File: packages/astro-imagetools/api/renderBackgroundPicture.js [hunk 1]
```diff
@@ -38,7 +38,6 @@ export default async function renderBackgroundPicture(props) {
     img: imgAttributes = {},
     link: linkAttributes = {},
     style: styleAttributes = {},
-    picture: pictureAttributes = {},
```
Comment (IanVS) [inline packages/astro-imagetools/api/renderBackgroundPicture.js:41]: I'm not sure if deleting this is correct, or if it should be used down below.

File: packages/astro-imagetools/api/renderPicture.js [hunk 2]
```diff
@@ -37,7 +37,6 @@ export default async function renderPicture(props) {
     img: imgAttributes = {},
     link: linkAttributes = {},
     style: styleAttributes = {},
-    picture: pictureAttributes = {},
```
Comment (IanVS) [inline packages/astro-imagetools/api/renderPicture.js:40]: I'm not sure if deleting this is correct, or if it should be used down below.

File: package.json [hunk 3]
```diff
@@ -2,6 +2,7 @@
   "private": "true",
   "scripts": {
     "format": "prettier . --write",
+    "lint": "eslint .",
```
Comment (RafidMuhymin) [inline package.json:5]: Can you please put the `lint` command before the `format` comment? It makes the formatting look a bit better.

Comment (IanVS) [reply package.json:4]: Done, haha, not sure how I put it there.

**Review by RafidMuhymin — APPROVED** [review]

**Pull request merged**